### DISCUSSION TITLE
Ruler configuration for S3/MinIO

### DISCUSSION THREAD

**Discussion by ChinmayMishra** [cortexproject/cortex#3716] [Q&A] [answered]
I tried configuring the rules using rule files in S3. But not able to make it work. I placed the rule files in a bucket/<Tanent ID/Org ID>/rules1.yaml. But the ruler is not picking it up.

**Comment by pracucci** [comment, marked as answer]
👋  @ChinmayMishra. The location of rules in the storage include the namespace too, so it's something like `bucket/<tenant ID>/<namespace>/rules.yaml`. However I would suggest you to not manually upload rules to the storage, but use the [appropriate API](https://cortexmetrics.io/docs/api/#set-rule-group). Please remember to enable to enable the API using `-experimental.ruler.enable-api` (they're still marked experimental but we're using it in production and works fine).

**Reply by ChinmayMishra** [reply]
Thanks a lot @pracucci. Using the API it worked. Earlier I was trying to upload the rule files manually to s3 and that is to the wrong folder.

**Reply by Rahuly360** [reply]
@ChinmayMishra @pracucci can you please provide me the Ruler's configuration to use s3 Storage. Also sample Rule rule and Experimental API. 
I am using sample rule:-
name: ruler_check_rules
rules:
  - alert: check_new_up
    expr: 'up==1'
    annotations:
      ruler: cortex
    labels:
      cortex: ruler

And API: curl -XPOST  -HContent-Type:application/yaml --data-binary @rules.yaml http://localhost:30681/api/v1/rules/default

But the rules are not getting triggered.

**Reply by pracucci** [reply]
I would suggest to use this tool to update ruler config to Cortex:
https://github.com/grafana/cortex-tools#cortextool

That being said, you can find an example of the rules config file here:
https://github.com/cortexproject/cortex/blob/master/development/tsdb-blocks-storage-s3/config/rules.yaml
